micro-rdk/src/common/app_client.rs

```diff
@@ -301,18 +245,13 @@ impl AppClient {
         };
 
         let cfg_response = ConfigResponse::decode(r.split_off(5))?;
-        if let Some(robot_config) = cfg_response.config.as_ref() {
```

Comment:
I do sort of see why it made sense to have the `rpc_host` be part of the `AppClient`, because without it `AppClient` doesn't manage all the state it needs to perform all the various tasks against app. However, I also don't see a way to keep mutation out of `AppClient`  if it is supposed to manage that state, since it only learns of it via the `get_config` method. It makes me wonder a bit if `get_config` belongs on `AppClient` at all, or more the responsibility of `AppClientBuilder`.

Reply:
I disagree, the idea of a host isn't really "client-side" behavior, so it makes sense that a client should not be in charge of the state of the host. It's the server that is using an application client to act as a host making it a better candidate for maintaining the state of something like `rpc_host`

Reply:
OK, I'm reasonably convinced.